test/arc_txt_endpoint_test.go

```diff
@@ -237,3 +238,174 @@ func TestHttpPost(t *testing.T) {
+	jsonPayload := fmt.Sprintf(`{"rawTx": "%s"}`, txHexString)
+	resp, err := postTx(t, jsonPayload, nil)
+	if err != nil {
+		t.Fatalf("Error sending HTTP request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 422 {
+		t.Errorf("Expected 422 Unprocessable Entity but got: %d", resp.StatusCode)
+	}
+}
+
+func TestPostTx_BadRequestBodyFormat(t *testing.T) {
+	// Instead of sending a valid hex string, send a JSON
+	improperPayload := `{"transaction": "fakeData"}`
+
+	resp, err := postTx(t, improperPayload, nil) // Using the helper function for the single tx endpoint
+	if err != nil {
+		t.Fatalf("Error sending HTTP request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected 400 Bad Request due to improper body format but got: %d", resp.StatusCode)
+	}
+}
+
+func TestPostTx_ConflictingTx(t *testing.T) {
+	txHexString := createTxHexStringExtended(t) // This is a placeholder for the method to create a valid transaction string.
+	conflictingtxt := createTxHexStringExtended(t) // This is a placeholder for the method to create a valid transaction string.
+
+	jsonPayload := fmt.Sprintf(`{"rawTx": "%s"}`, txHexString)
+	resp, err := postTx(t, jsonPayload, nil) // no extra headers
+	if err != nil {
+		t.Fatalf("Error sending HTTP request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	jsonPayload := fmt.Sprintf(`{"rawTx": "%s"}`, conflictingtxt)
+	resp, err := postTx(t, jsonPayload, nil) // no extra headers
+	if err != nil {
+		t.Fatalf("Error sending HTTP request: %s", err)
+	}
+	defer resp.Body.Close()
+
+
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("Expected 200 OK but got: %d", resp.StatusCode)
+	}
+}
+
+	if resp.StatusCode != 466 {
+		t.Errorf("Expected 466 Conflicting transaction found but got: %d", resp.StatusCode)
+	}
+}
+
+func postTx(t *testing.T, jsonPayload string, headers map[string]string) (*http.Response, error) {
+	url := "http://arc:9090/v1/tx"
```

Comment:
maybe worth extracting the arc IP to env variable to be able to run it outside docker container?

Reply:
Good idea. I will put that in a separate PR for enhanchments as the other tests are currently working in this way.